Source/EasyNetQ/Internals/TaskHelpers.cs

```diff
@@ -15,23 +15,7 @@ public static class TaskHelpers
     ///     doing so can result in application failures when updating to a new EasyNetQ release.
     /// </summary>
     public static Func<T1, CancellationToken, Task<T2>> FromFunc<T1, T2>(Func<T1, CancellationToken, T2> func)
-    {
-        return (x, c) =>
-        {
-            try
-            {
-                return Task.FromResult(func(x, c));
-            }
-            catch (OperationCanceledException oce)
-            {
-                return Task.FromCanceled<T2>(oce.CancellationToken);
-            }
-            catch (Exception exception)
-            {
-                return Task.FromException<T2>(exception);
-            }
-        };
-    }
+        => (x, c) => Task.Run(() => func(x, c), default);
```

Comment:
`=> (x, c) => Task.Run(() => func(x, c), default);`
->
`=> (x, c) => Task.Run(() => func(x, c), c);`

here and for other changed methods. Otherwise, the created task loses its connection with the cancellation token to which should react.

Reply:
It doesn't seem to matter so much, because a task here could only be cancelled if it has not yet started, which has very limited timeframe, because it should be scheduled shortly.

Reply:
>  should be scheduled shortly

In theory - yes. In practice it can be just another place of "interesting" bugs in the future. Better safe than sorry.

Reply:
Also cancellation token from the created task (None) will not be equal one from `TaskCancelledException` so some consumer logic may fail if it compares tokens (in catch filters or whereever).

Reply:
https://github.com/EasyNetQ/EasyNetQ/commit/ebd1f9ce27668bbf4c8db530d950ef5b7714a932